common/src/main/java/com/microsoft/identity/common/internal/cache/DefaultSharedPrefsFileManagerReencrypter.java

```diff
@@ -0,0 +1,96 @@
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.cache;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.internal.controllers.TaskCompletedCallbackWithError;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Default implementation of {@link ISharedPrefsFileManagerReencrypter}.
+ */
+public class DefaultSharedPrefsFileManagerReencrypter implements ISharedPrefsFileManagerReencrypter {
+
+    @Override
+    public void reencrypt(@NonNull final ISharedPreferencesFileManager fileManager,
+                          @NonNull final IStringEncrypter encrypter,
+                          @NonNull final IStringDecrypter decrypter,
+                          @NonNull final ReencryptionParams params) throws Exception {
+        final Map<String, String> cacheEntries = new HashMap<>(fileManager.getAll());
```

Comment:
I hate to ask, but what's guarding access while all this is going on?  I think it's all up to use behavior, make sure that while you are running the re-encryption, you don't insert any new keys?

Reply:
You got it -- this is going to have to be guidanced-based. There's currently no way to programmatically enforce we have exclusive domain over the underlying `SharedPreferences` instance

Reply:
Do you want to be really explicit about this in the API documentation, then?

Reply:
Roger wilco

Reply:
69c7eb6a46aefd9fd3b96d21ca9606ff15536d3e